File: apps/api/src/kimy/schemas/stats.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class StatusCountOut(BaseModel):
    status: str
    count: int


class ProgramGradeOut(BaseModel):
    program_id: str
    program_code: str
    program_name: str
    average_grade: float
    submissions_count: int


class StatsOverviewOut(BaseModel):
    total_submissions: int
    total_advisors_with_orcid: int
    submissions_by_status: list[StatusCountOut]
    avg_ai_grade: float | None
    avg_ai_percentage: float | None
    ai_human_concordance_pct: float | None
    plagiarism_alerts: int
    advisor_fit_alerts: int
    low_compliance_submissions: int
    citations_total: int
    citations_problematic: int
    grades_per_program: list[ProgramGradeOut]


class ActivityItemOut(BaseModel):
    kind: str
    occurred_at: datetime
    submission_id: str
    submission_title: str
    actor_name: str
    description: str

File: apps/web/src/lib/api/stats.ts
"use server";

import { apiFetch } from "@/lib/api/client";

export type StatusCount = { status: string; count: number };
export type ProgramGrade = {
  program_id: string;
  program_code: string;
  program_name: string;
  average_grade: number;
  submissions_count: number;
};
export type StatsOverview = {
  total_submissions: number;
  total_advisors_with_orcid: number;
  submissions_by_status: StatusCount[];
  avg_ai_grade: number | null;
  avg_ai_percentage: number | null;
  ai_human_concordance_pct: number | null;
  plagiarism_alerts: number;
  advisor_fit_alerts: number;
  low_compliance_submissions: number;
  citations_total: number;
  citations_problematic: number;
  grades_per_program: ProgramGrade[];
};
export type ActivityItem = {
  kind: string;
  occurred_at: string;
  submission_id: string;
  submission_title: string;
  actor_name: string;
  description: string;
};

export async function fetchStatsOverview(programId?: string): Promise<StatsOverview> {
  const qs = programId ? `?program_id=${programId}` : "";
  return apiFetch<StatsOverview>(`/api/v1/stats/overview${qs}`);
}

export async function fetchStatsActivity(
  limit = 15,
  programId?: string,
): Promise<ActivityItem[]> {
  const params = new URLSearchParams({ limit: String(limit) });
  if (programId) params.set("program_id", programId);
  return apiFetch<ActivityItem[]>(`/api/v1/stats/activity?${params.toString()}`);
}
